### DISCUSSION TITLE
[Core Protocol] Enable Gas Estimate On Wallet Apps [Metamask]

### DISCUSSION THREAD

**Discussion by avenbreaks** [ethereum-optimism/developers#619] [Core Protocol]
### Did you check the documentation and specs?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Please select the type of request

- [ ] Bug Report
- [ ] Feature Request

### Operating System

Ubuntu

### Relevant Context or Examples

How to enable gasfee estimation like in metamask / blockscout view. where there are 3 options : 
Low / Slow 
Market / Normal 
Aggressive / Fast

This is from Node Configuration or what ? 

Thanks. 

![image](https://github.com/user-attachments/assets/48eeb5ec-6952-4d01-bc0e-5c050a991113)

**Comment by hash1go** [comment]
Hello.
I haven't examined the Metamask extension code, but generally, the recommended gas price is calculated as follows:

* The app queries the RPC
* The RPC returns a gas price
* Based on that gas price, it calculates and presents a recommended gas price to the user

In other words, the function of returning gas prices through HTTP communication is not included in the protocol itself, but it's a feature arbitrarily implemented by clients or third-party RPC providers.
